clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -123,6 +123,10 @@ class Util {
   /// specialization id class.
   static bool isSyclSpecIdType(QualType Ty);
 
+  /// Checks whether given clang type is a full specialization of the SYCL
+  /// device_global class.
+  static bool isSyclDeviceGlobalType(QualType Ty);
```

Comment:
Just FYI, I refactored isSyclDeviceGlobalType to be a member of Sema 
https://github.com/intel/llvm/pull/5597/commits/df337a7de8f9164bc63c7d82ed6e0f6120920aa5#diff-edac6256ac508912a16d0165b2f8cf37123dc2f40a147dca49a34c33f1db13dd